src/java/org/apache/cassandra/index/sai/memory/MemtableIndex.java

```diff
@@ -79,7 +82,7 @@ public ByteBuffer getMaxTerm()
 
     public long index(DecoratedKey key, Clustering<?> clustering, ByteBuffer value)
     {
-        if (value == null || value.remaining() == 0)
+        if (value == null || (!type.allowsEmpty() && value.remaining() == 0))
```

Comment:
```suggestion
        if (value == null || (value.remaining() == 0 && !type.allowsEmpty()))
```

nit: we only want to call `allowsEmpty` when its empty, else it adds a virtual dispatch that is `false` in most cases... so slightly less overhead in the common case (non-empty)